core/transaction.go

```diff
@@ -70,20 +72,65 @@ type Transaction interface {
 }
 
 var (
-	_ Transaction = (*DeployTransaction)(nil)
-	_ Transaction = (*DeployAccountTransaction)(nil)
-	_ Transaction = (*DeclareTransaction)(nil)
-	_ Transaction = (*InvokeTransaction)(nil)
-	_ Transaction = (*L1HandlerTransaction)(nil)
+	_            Transaction = (*DeployTransaction)(nil)
+	_            Transaction = (*DeployAccountTransaction)(nil)
+	_            Transaction = (*DeclareTransaction)(nil)
+	_            Transaction = (*InvokeTransaction)(nil)
+	_            Transaction = (*L1HandlerTransaction)(nil)
+	queryVersion             = new(felt.Felt).Exp(new(felt.Felt).SetUint64(2), new(big.Int).SetUint64(queryBit))
 )
 
 const (
 	// Calculated at https://hur.st/bloomfilter/?n=1000&p=&m=8192&k=
 	// provides 1 in 51 possibility of false positives for approximately 1000 elements
 	eventsBloomLength    = 8192
 	eventsBloomHashFuncs = 6
+	queryBit             = 128
 )
 
+type TransactionVersion felt.Felt
```

Comment:
Should we add a note here that we need a migration if we change the underlying type of `TransactionVersion`?

Reply:
Will do but I think one should already be aware that all `core` types are used as storage objects and be careful when making any changes on them.